packages/starknet-snap/src/rpcs/sign-declare-transaction.ts

```diff
@@ -0,0 +1,120 @@
+import type { Component } from '@metamask/snaps-sdk';
+import {
+  heading,
+  row,
+  text,
+  UserRejectedRequestError,
+} from '@metamask/snaps-sdk';
+import type { DeclareSignerDetails } from 'starknet';
+import type { Infer } from 'superstruct';
+import { array, object, string, assign } from 'superstruct';
+
+import {
+  confirmDialog,
+  AddressStruct,
+  toJson,
+  BaseRequestStruct,
+  AccountRpcController,
+  DeclareSignDetailsStruct,
+} from '../utils';
+import { signDeclareTransaction as signDeclareTransactionUtil } from '../utils/starknetUtils';
+
+export const SignDeclareTransactionRequestStruct = assign(
+  object({
+    address: AddressStruct,
+    details: DeclareSignDetailsStruct,
+  }),
+  BaseRequestStruct,
+);
+
+export const SignDeclareTransactionResponseStruct = array(string());
+
+export type SignDeclareTransactionParams = Infer<
+  typeof SignDeclareTransactionRequestStruct
+>;
+
+export type SignDeclareTransactionResponse = Infer<
+  typeof SignDeclareTransactionResponseStruct
+>;
+
+/**
+ * The RPC handler to sign a declare transaction.
+ */
+export class SignDeclareTransactionRpc extends AccountRpcController<
+  SignDeclareTransactionParams,
+  SignDeclareTransactionResponse
+> {
+  protected requestStruct = SignDeclareTransactionRequestStruct;
+
+  protected responseStruct = SignDeclareTransactionResponseStruct;
+
+  /**
+   * Execute the sign declare transaction request handler.
+   * It will show a confirmation dialog to the user before signing the declare transaction.
+   *
+   * @param params - The parameters of the request.
+   * @param params.address - The address of the signer.
+   * @param params.details - The declare transaction details to sign.
+   * @param [params.enableAuthorize] - Optional, a flag to enable or display the confirmation dialog to the user.
+   * @param params.chainId - The chain id of the network.
```

Comment:
Why do we have chainId in here ? Isn't this already in this.network.chainId

Reply:
this has to set up as the parent class is using the chainId, please refer to the parent class to know more